File: script/ssh_try.go
//go:build ignore

package main

import (
	"fmt"
	"os"
	"time"

	"golang.org/x/crypto/ssh"
)

func main() {
	if len(os.Args) < 4 {
		fmt.Println("用法: go run script\\ssh_try.go <host:port> <user> <password>")
		os.Exit(1)
	}
	addr, user, pass := os.Args[1], os.Args[2], os.Args[3]

	cfg := &ssh.ClientConfig{
		User:            user,
		Auth:            []ssh.AuthMethod{ssh.Password(pass)},
		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
		Timeout:         8 * time.Second,
	}

	conn, err := ssh.Dial("tcp", addr, cfg)
	if err != nil {
		fmt.Printf("FAIL: %v\n", err)
		os.Exit(1)
	}
	defer conn.Close()
	fmt.Println("OK: connection established")

	// 执行 ls 命令
	session, err := conn.NewSession()
	if err != nil {
		fmt.Println("new session FAIL:", err)
		os.Exit(1)
	}
	defer session.Close()

	out, err := session.Output("ls")
	if err != nil {
		fmt.Println("exec ls FAIL:", err)
		os.Exit(1)
	}
	fmt.Println("ls 输出:")
	fmt.Println(string(out))
}

File: script/ssh_try.py
# -*- coding: utf-8 -*-
"""SSH 连接测试脚本：python ssh_try.py host port user password"""
import sys
import paramiko


def main():
    if len(sys.argv) < 5:
        print("用法: python ssh_try.py <host> <port> <user> <password>")
        sys.exit(1)
    host, port, user, password = sys.argv[1], int(sys.argv[2]), sys.argv[3], sys.argv[4]

    client = paramiko.SSHClient()
    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
    try:
        client.connect(host, port=port, username=user, password=password, timeout=8)
        stdin, stdout, stderr = client.exec_command("ls")
        print("OK: connection established")
        print("ls 输出:", stdout.read().decode("utf-8", "ignore").strip())
    except Exception as e:
        print("FAIL:", e)
        sys.exit(1)
    finally:
        client.close()


if __name__ == "__main__":
    main()
